library/psa_crypto.c

```diff
@@ -6245,44 +6245,75 @@ static psa_status_t psa_key_derivation_setup_kdf(
     if (!is_kdf_alg_supported(kdf_alg)) {
         return PSA_ERROR_NOT_SUPPORTED;
     }
+    psa_status_t status = PSA_SUCCESS;
+
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_PMS)
+    if (kdf_alg == PSA_ALG_TLS12_ECJPAKE_TO_PMS) {
+        operation->capacity = PSA_HASH_LENGTH(PSA_ALG_SHA_256);
+        return PSA_SUCCESS;
```

Comment:
Returning early here could be error-prone if we want to add more things than setting the capacity. So please create a separate function `psa_key_derivation_set_maximum_capacity`.

Reply:
Can you elaborate on the function `psa_key_derivation_set_maximum_capacity`

Reply:
I think it would be best to have an internal `psa_key_derivation_set_maximum_capacity` whose sole job is to set `operation->capacity`, or return an error if it's given an algorithm that it doesn't support.